### PR TITLE
Update react-navigation.md

### PR DESCRIPTION
**Pull request by ThaJay**
Update docs to reflect react-navigation 5x with hooks

### CODE REVIEW

**Comment by MarcoScabbiolo** [comment]
While updating the example to use react-navigation 5 is ok, the example is wrong because you cannot use a hook inside a class.

File: docs/react-navigation.md [hunk 1]
```diff
@@ -8,10 +8,10 @@ React-navigation does not unmount components when switching between tabs. So whe
 
 ```jsx
 
-import { withNavigationFocus } from 'react-navigation' 
+import { useIsFocused } from '@react-navigation/core';
 
  render() {
-    const { isFocused } = this.props
+    const isFocused = useIsFocused();
```
Comment (MarcoScabbiolo) [inline docs/react-navigation.md:14]: Hooks cannot be used in a class component. Making the example a function component would be ideal.

**Review by MarcoScabbiolo — CHANGES_REQUESTED** [review]

File: docs/react-navigation.md (hunk 1 above)
Comment (ThaJay) [reply docs/react-navigation.md:14]: Ah yes I missed that. Will do.

Comment (ThaJay) [reply docs/react-navigation.md]: I think I have changed it properly now.

**Review by MateusAndrade — APPROVED** [review]

**Review by MarcoScabbiolo — APPROVED** [review]

**Pull request merged**